cyclops/tasks/cxr_classification.py

```diff
@@ -0,0 +1,323 @@
+"""Chest X-ray Classification Task."""
+import logging
+from functools import partial
+from typing import Any, Dict, List, Optional, Sequence, Tuple, Union, get_args
+
+import numpy as np
+from datasets import Dataset
+from monai.transforms import Compose
+
+from cyclops.datasets.slicing import SlicingConfig as SliceSpec
+from cyclops.evaluate.evaluator import Evaluator
+from cyclops.evaluate.metrics.metric import Metric, MetricCollection, create_metric
+from cyclops.models.catalog import _img_model_keys
+from cyclops.models.utils import get_device
+from cyclops.models.wrappers import WrappedModel
+from cyclops.tasks.util import apply_image_transforms
+from cyclops.utils.log import setup_logging
+
+# from cyclops.datasets.slice import SliceSpec XXX: Add when the branch is merged
+
+
+LOGGER = logging.getLogger(__name__)
+setup_logging(print_level="INFO", logger=LOGGER)
+
+
+class CXRClassification:
+    """Chest X-ray classification task modeled as a multi-label classification task."""
+
+    def __init__(
+        self,
+        models: Union[WrappedModel, Sequence[WrappedModel], Dict[str, WrappedModel]],
+        task_features: List[str],
+        task_target: str,
+    ):
+        """Chest X-ray classification task.
+
+        Parameters
+        ----------
+        models : Union[WrappedModel, Sequence[WrappedModel], Dict[str, WrappedModel]]
+            The model(s) to be used for prediction, and evaluation.
+        task_features : List[str]
+            List of feature names.
+        task_target : str
+            List of target names.
+
+        """
+        self.models = self._prepare_models(models)
+        self.task_features = task_features
+        self.task_target = task_target
+        self.evaluator = Evaluator()
+        self.device = get_device()
+
+    @property
+    def task_type(self):
+        """The classification task type.
+
+        Returns
+        -------
+        str
+            Classification task type.
+
+        """
+        return "multilabel"
+
+    @property
+    def data_type(self):
+        """The data type.
+
+        Returns
+        -------
+        str
+            The data type.
+
+        """
+        return "image"
+
+    def _prepare_models(
```

Comment:
Would be good to abstract this into a utility method that is implemented somewhere else. That way every other task (and even the evaluator) can use it without redefining it.

Reply:
Can do that for the task. But does the evaluator want to only support the wrapped and registered model? since the task _prepare_model also checks for the type of the registered model

Reply:
Yes. Internally, we will use the wrapped models. But we should accept unwrapped models as input and try to wrap it - something the `prepare_models` method can handle.